packages/kbn-reporting/export_types/csv/csv_v2.ts

```diff
@@ -87,14 +87,10 @@ export class CsvV2ExportType extends ExportType<
       throw Boom.badRequest('Invalid Job params: must contain a single Discover App locator');
     }
 
-    if (!params || !params.savedSearchId || typeof params.savedSearchId !== 'string') {
```

Comment:
For the non es|ql reports csv_v2 still needs the savedSearchId. this is asserted deeper inside the code branch. 
 
es|ql reports are working without savedSearchId

full support and alignment for saved and non saved searches should be implemented in https://github.com/elastic/kibana/issues/151190

Reply:
++ on the idea to implement ES|QL CSV export within the csv_v2 export type.